### PR TITLE
Move cbmc_dimacs to solvers/flattening/bv_dimacs

### PR DESCRIPTION
**Pull request by peterschrammel**
because it can be used independently of CBMC.

Based on https://github.com/diffblue/cbmc/pull/2881

### CODE REVIEW

**Review by tautschnig — CHANGES_REQUESTED** [review]
Looks mostly good, except I believe this should go in solvers/sat, not solvers/flattening.

**Comment by peterschrammel** [comment]
> this should go in solvers/sat, not solvers/flattening.

It requires a bit blasting backend to make sense (note that it outputs word/bit-level variable mappings).

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
It seems we anyway already had some dependencies from flattening/ -> sat/.

**Comment by peterschrammel** [comment]
Let me check -  'anyway already had' is not a good reason for doing something...

**Comment by peterschrammel** [comment]
> It seems we anyway already had some dependencies from flattening/ -> sat/.

No, this is clean; the dependency is only in one direction.

File: src/cbmc/cbmc_solvers.cpp [hunk 1]
```diff
@@ -93,8 +93,8 @@ std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_dimacs()
 
   std::string filename=options.get_option("outfile");
 
-  auto cbmc_dimacs=util_make_unique<cbmc_dimacst>(ns, *prop, filename);
-  return util_make_unique<solvert>(std::move(cbmc_dimacs), std::move(prop));
+  auto bv_dimacs=util_make_unique<bv_dimacst>(ns, *prop, filename);
```
Comment (mgudemann) [inline src/cbmc/cbmc_solvers.cpp]: nitpick run `clang-format`

**Review by mgudemann — APPROVED** [review]

**Pull request merged**